packages/cadl-ranch-specs/http/azure/client-generator-core/flatten/main.tsp

```diff
@@ -0,0 +1,107 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Illustrates the model flatten cases.")
+@scenarioService("/type/model/flatten")
+namespace Type.Model.Flatten;
+
+@doc("This is the model with one level of flattening.")
+model FlattenModel {
+  name: string;
+
+  #suppress "deprecated" "@flattenProperty decorator is not recommended to use."
+  @flattenProperty
+  properties: ChildModel;
+}
+
+@doc("This is the model with two levels of flattening.")
+model NestedFlattenModel {
+  name: string;
+
+  #suppress "deprecated" "@flattenProperty decorator is not recommended to use."
+  @flattenProperty
+  properties: ChildFlattenModel;
+}
+
+@doc("This is the child model to be flattened.")
+model ChildModel {
+  description: string;
+  age: int32;
+}
+
+@doc("This is the child model to be flattened. And it has flattened property as well.")
+model ChildFlattenModel {
+  summary: string;
+
+  #suppress "deprecated" "@flattenProperty decorator is not recommended to use."
+  @flattenProperty
+  properties: ChildModel;
+}
+
+@scenario
+@route("/flattenModel")
+@scenarioDoc("""
+Update and receive model with 1 level of flattening.
+Expected input body:
+```json
+{
+  "name": "foo",
+  "properties": {
+    "description": "bar",
+    "age": 10
+  }
+}
+```
+
+Expected response body:
+```json
+{
+  "name": "test",
+  "properties": {
+    "description": "test",
+    "age": 1
+  }
+}
```

Comment:
nit, there seems no reason to have response json be different from request json?

Reply:
yes. I intentionally make it different so that people cannot be lazy when composing test cases in each language...